src/shared/pager.c

```diff
@@ -147,10 +152,14 @@ int pager_open(bool no_pager, bool jump_to_end) {
         }
 
         /* Return in the parent */
+        stored_stdout = fcntl(STDOUT_FILENO, F_DUPFD_CLOEXEC, 3);
+        stored_stderr = fcntl(STDERR_FILENO, F_DUPFD_CLOEXEC, 3);
         if (dup2(fd[1], STDOUT_FILENO) < 0)
                 return log_error_errno(errno, "Failed to duplicate pager pipe: %m");
+        stdout_redirected = true;
         if (dup2(fd[1], STDERR_FILENO) < 0)
                 return log_error_errno(errno, "Failed to duplicate pager pipe: %m");
```

Comment:
I'd prefer if we'd roll back initialization of stored_stdout/stored_stderr here (i think you had that in a previous version of the patch, no?). we generally try to write our functions to have as little side-effects as possible if they fail. (It's not always possible to sanely rollback every step, but in this case it is, hence we should do it).